secure-boot-example/README.md

```diff
@@ -1,8 +1,153 @@
+[secure-boot-recovery/README](../secure-boot-recovery/README.md] guides for
+more information.
+
+### Requirements for running this example
+* A Raspberry Pi Compute Module 4 which has not had secure-boot enabled in OTP.
+* Micro USB cable for `rpiboot` connection
+* USB serial cable (for debug logs)
+* Linux or Cygwin (Windows 10)
+* OpenSSL
+* Python3
+* Python `cryptodomex`
 
-To sign the image with a different key run
 ```bash
+python3 -m pip install pycryptodomex
+# or
+pip install pycryptodomex
+```
+
+### Clean configuration
+Before starting it's advisible to create a fresh clone of the `usbboot` repo
+to ensure that there are no stale configuration files.
+
+```bash
+git clone https://github.com/raspberrypi/usbboot secure-boot
+cd secure-boot
+make
+```
+See the top-level [README](../Readme.md) for build instructions.
+
+### Hardware setup for `rpiboot` mode
+Prepare the Compute Module for `rpiboot` mode.
+
+* Set the `nRPIBOOT` jumper which is labelled `Fit jumper to disable eMMC Boot' on the Compute Module 4 IO board.
+* Connect the micro USB cable to the `USB slave` port on the CM4 IO board.
+* Power cycle the CM4 IO board.
+* Connect the USB serial adapter to [GPIO 14/15](https://www.raspberrypi.com/documentation/computers/os.html#gpio-and-the-40-pin-header) on the 40-pin header.
+
+### Reset the Compute Module EEPROM
+Enable `rpiboot` mode and flash the latest EEPROM image
+```bash
+./rpiboot -d recovery
+```
+
+### Boot the example image
+Enable `rpiboot` mode and load the OS via `rpiboot` without enabling code-signing.
+```bash
+./rpiboot -d secure-boot-example
+```
+The OS should load and show activity on the boot UART and HDMI console.
+
+### Generate a signing key
+Secure-boot requires a 2048 bit RSA private key. You can either use a pre-existing
+key or generate an specific key for this example. The `KEY_FILE` environment variable
+used in the following instructions must contain the absolute path to the RSA private key in
+PEM format.
+```bash
+openssl genrsa 2048 > private.pem
+export KEY_FILE=$(pwd)/private.pem
+```
+
+### Update the EEPROM to require signed OS images
+Enable `rpiboot` mode and flash the bootloader EEPROM with updated setting enables code-signing.
+
+The `boot.conf` file sets the `SIGNED_BOOT` property `1` which instructs the bootloader to only
+load files from `boot.img` and verify this against the signature in `boot.sig`. The `update-pieeprom.sh`
+generates the signed `pieeprom.bin` image.
+
+```bash
+cd secure-boot-recovery
+# Generate the signed EEPROM image.
+../tools/update-pieeprom.sh -k "${KEY_FILE}"
+cd ..
+./rpiboot -d secure-boot-recovery
+```
+
+At this stage the signed image requirement can be reverted by flashing a default, unsigned image. 
+However, once the [OTP secure-boot flags](../secure-boot-recovery/README.md#locking-secure-boot-mode) are set then `SIGNED_BOOT` is permenantely enabled and cannot be overriden via the EEPROM config.
```

Comment:
I would clarify that we're not enabling secure boot in these instructions?

Reply:
tweaked first sentence